### ISSUE TITLE
Figure out how to handle files that should be generated but already exist with `writeToCache`

### ISSUE THREAD

**Issue by natebosch** [dart-lang/build#390]
Now that we can generate files to a cache directory without overwriting the same file in a source directory we need to figure out how we should react.

We could throw an error and refuse to attempt the build.

We could build anyway and make sure reads go to the generated version, ignoring the source version altogether. I think this is what we'd do today anyway but we should formalize this with tests if it's the behavior we want.

**Comment by jakemac53** [comment]
imo we should keep the original behavior, if connected to a console then prompt the user to delete the files (or delete them right away if they have that option set),  otherwise error and don't attempt the build.

**Comment by jakemac53** [comment]
or actually if it is known to be a previously built file because it exists in the cache directory, then we could skip the prompt altogether and just delete it (its safe to assume its from a previous build).

**Comment by natebosch** [comment]
For files in external packages we can't delete them.

Synced offline and we want to treat these as errors - we shouldn't let you overwrite a file that exists in sources. Here's some stuff that we want to do - recording here for now and might split these out into separate issues:

- [ ] Error when a generated file would overwrite a Source (which isn't in the cache)
- [ ] Skip the prompt for deleting files from the cache, we always know this is safe
- [ ] Add a script hash into the generated directory path so different builds can have different generated files (though we expect our automatic one to be the only one using the cache)
- [ ] When we get to the point of generating the script, combine `build` and `watch` into one entry point with different arguments so they share the same hash.

**Comment by matanlurey** [comment]
Just to clarify, is this to deal with builders not using `writeToCache: true` today?

**Comment by jakemac53** [comment]
Yes, the biggest pain point will be if one of them switches to write to cache.

**Comment by matanlurey** [comment]
@jakemac53:

> Yes, the biggest pain point will be if one of them switches to write to cache.

Do we expect this to be an issue for the MVP?

**Comment by jakemac53** [comment]
> Do we expect this to be an issue for the MVP?

Not really, no. We can always re-evaluate if people run into issues though.

**Comment by matanlurey** [comment]
Moving to M1!
